test/epipe_tests.erl

```diff
@@ -59,9 +59,27 @@ pipe_test_() ->
             Fun1 = fun(Val) -> {ok, Val + 1} end,
             Fun2 = fun(_Val) -> {error, "Can't process data"} end,
 
+            ?assertEqual(
+                {error, step2, "Can't process data", 1},
+                epipe:run([{step1, Fun1}, {step2, Fun2}, {step3, Fun1}], 0)
+            )
+        end},
+        {"Function raises an error", fun() ->
+            Fun1 = fun(Val) -> {ok, Val + 1} end,
+            Fun2 = fun(_Val) -> erlang:error("Bad CRC")  end,
+
+            ?assertEqual(
+                {error, step2, "Can't process data", 1},
+                epipe:run([{step1, Fun1}, {step2, Fun2}, {step3, Fun1}], 0)
+            )
+        end},
+        {"Function throws an error", fun() ->
+            Fun1 = fun(Val) -> {ok, Val + 1} end,
+            Fun2 = fun(_Val) -> throw("Terrible CRC")  end,
```

Comment:
This does not follow the spec of the run function https://github.com/oltarasenko/epipe/blob/master/src/epipe.erl#L10-L19. See also http://erlang.org/doc/man/erlang.html#exit-1